### PR TITLE
Improve Version Regression Check

### PR DESCRIPTION
**Pull request by danworth**
Allow earlier versions to be deployed if they are the latest enabled version. This allows developers to deploy earlier versions by disabling any later versions whilst protecting from unintended regression due to Concourse quirks.

### HOW TO TEST ###
Can be run locally with the following, adjusting env vars to prompt various behaviour. Log in first locally with `fly login --target pay-dev`

```
CONTAINER_SUFFIX=test PIPELINE_NAME=deploy-to-test FLY_USERNAME=pay-dev APP_NAME=adminusers CLUSTER_NAME=test-12-fargate APPLICATION_IMAGE_TAG=1919 aws-vault exec pay-test-admin -- node check-release-versions.js


You are trying to deploy release  1919 of adminusers which is older than the current release 1920.

Checking that 1919 is the latest ENABLED release.
The latest ENABLED release is 1920
1919 of adminusers is not the latest ENABLED version. Bailing out.
If you need to deploy this version you must disable later versions (pinning will not work)

```
and for example
```
➤➤ scripts $ CONTAINER_SUFFIX=test PIPELINE_NAME=deploy-to-test FLY_USERNAME=pay-dev APP_NAME=adminusers CLUSTER_NAME=test-12-fargate APPLICATION_IMAGE_TAG=1919 aws-vault exec pay-test-admin -- node check-release-versions.js


You are trying to deploy release  1919 of adminusers which is older than the current release 1920.

Checking that 1919 is the latest ENABLED release.
The latest ENABLED release is 1919
Proceeding to deploy release 1919
```

Please help by checking for typos, it seems a little brittle in the reliance on config values between the pipeline and the node task. I have tried to consider various failure scenarios with helpful error messages. This new logic should only be invoked when someone attempts to deploy an older version. There is retry logic because sometimes the fly check resources command does not return any results...

### CODE REVIEW

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-dev/bastion.yml
pay-dev/cardid-s3.yml
pay-dev/deploy-to-perf.yml
pay-dev/e2e-helpers.yml
pay-dev/infra-drift-detector.yml
pay-dev/init-lock-pools.yml
pay-dev/perf-tests.yml
pay-dev/pkl-pipeline-changes.yml
pay-dev/pr-ci.yml
```

**Comment by alphagov-pay-ci-concourse** [comment]
The diff for `pay-deploy/pay-deploy/deploy-to-production.pkl` was too long, see concourse output

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-deploy/bastion.yml
pay-deploy/concourse-resources.yml
pay-deploy/concourse-runner.yml
pay-deploy/detect-secrets.yml
pay-deploy/infra-drift-detector.yml
pay-deploy/init-lock-pools.yml
pay-deploy/internal-vulnerability-scan.yml
pay-deploy/node-runner.yml
pay-deploy/pact-broker.yml
pay-deploy/pay-cli.yml
pay-deploy/pay-js.yml
pay-deploy/pkl-pipeline-changes.yml
pay-deploy/prometheus-pushgateway.yml
```

File: ci/scripts/check-release-versions.js [hunk 1]
```diff
@@ -36,11 +37,122 @@ function checkReleaseVersion (containerName, tagToBeDeployed, currentContainerDe
...
+
+  do {
+    tryCount += 1
+    const releases = await getReleases(resource, flyTeam)
+    latestEnabledRelease = releases.find((release) => release.enabled)
+
+    if (!latestEnabledRelease) {
+      console.log(`There are no enabled releases for ${resource}, sometimes this is due to flakiness in the fly api call`)
+      console.log(`Retrying, attempt ${tryCount} of ${MAX_ATTEMPTS} in ${RETRY_WAIT_TIME_SECONDS} seconds`)
+      await new Promise(resolve => setTimeout(resolve, RETRY_WAIT_TIME_SECONDS * 1000))
```
Comment (jfharden) [inline ci/scripts/check-release-versions.js:91]: There's a nicer way to sleep these days

```
const { setTimeout } = require('node:timers/promises');

await setTimeout(1000)
```

If you need setTimeout and the promise version you can

```
const { setTimeout: setTimeoutPromise } = require('node:timers/promises');

await setTimeoutPromise(1000)
```

**Review by jfharden — APPROVED** [review]
Just a thing that might make your future life easier, but happy with it anyway 👍

**Comment by alphagov-pay-ci-concourse** [comment]
The diff for `pay-deploy/pay-deploy/deploy-to-production.pkl` was too long, see concourse output

**Comment by alphagov-pay-ci-concourse** [comment]
The diff for `pay-deploy/pay-deploy/deploy-to-staging.pkl` was too long, see concourse output

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-deploy/bastion.yml
pay-deploy/concourse-resources.yml
pay-deploy/concourse-runner.yml
pay-deploy/detect-secrets.yml
pay-deploy/infra-drift-detector.yml
pay-deploy/init-lock-pools.yml
pay-deploy/internal-vulnerability-scan.yml
pay-deploy/node-runner.yml
pay-deploy/pact-broker.yml
pay-deploy/pay-cli.yml
pay-deploy/pay-js.yml
pay-deploy/pkl-pipeline-changes.yml
pay-deploy/prometheus-pushgateway.yml
```

**Comment by alphagov-pay-ci-concourse** [comment]
The diff for `pay-dev/pay-dev/deploy-to-test.pkl` was too long, see concourse output

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-dev/bastion.yml
pay-dev/cardid-s3.yml
pay-dev/deploy-to-perf.yml
pay-dev/e2e-helpers.yml
pay-dev/infra-drift-detector.yml
pay-dev/init-lock-pools.yml
pay-dev/perf-tests.yml
pay-dev/pkl-pipeline-changes.yml
pay-dev/pr-ci.yml
```

**Review by jfharden — APPROVED** [review]

**Pull request merged**